control-plane/subcommand/gateway-resources/command.go

```diff
@@ -288,6 +301,35 @@ func (c *Command) validateFlags() error {
 	return nil
 }
 
+func (c *Command) loadConfig() error {
+	// Load resources.json
+	file, err := os.Open("/consul/config/resources.json")
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		c.UI.Info("No resources.json found, using defaults")
+		c.resources = defaultResourceRequirements()
+		return nil
+	}
+
+	resources, err := io.ReadAll(file)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Unable to read resources.json, using defaults: %s", err))
+		c.resources = defaultResourceRequirements()
+		return err
+	}
```

Comment:
Looks like you designed this in a way that customers using a newer version of consul-k8s-control plane with an older version of the Helm chart will just get the default requirements ❤️

Reply:
Exactly! It will fall back nicely.